internal/config/config.go

```diff
@@ -169,16 +184,14 @@ func (conf *Config) SetPreferredPipelines(pipelines []pipeline.Pipeline) error {
 	return conf.localConfig.WriteConfig()
 }
 
-func firstNonEmpty[T comparable](t ...T) T {
-	var empty T
-
-	for _, k := range t {
-		if k != empty {
+func firstNonEmpty(s ...string) string {
+	for _, k := range s {
+		if k != "" {
```

Comment:
Was there a reason for changing this? I think we only ever give it strings, but in its current form it could be used with any comparable type.

Reply:
nothing other than YAGNI - i figured if we weren't using generics we probably shouldn't use generics, but i'm more than happy to change it back if you like

Reply:
Nah, not at all, think it was in as a just in case, but we can re-visit if it's ever needed